### PR TITLE
refactor: add most of src/util to iwyu

### PR DESCRIPTION
**Pull request by fanquake**
These files change infrequently, and not much header shuffling is required.

We don't add everything in src/util/ yet, because IWYU makes some
dubious suggestions, which I'm going to follow up with upstream.

Soon we'll swap `src/util/xyz.cpp` for just `src/util/`.

### CODE REVIEW

**Comment by hebasto** [comment]
Concept ACK.

File: ci/test/06_script_b.sh [hunk 1]
```diff
@@ -46,6 +46,18 @@ if [ "${RUN_TIDY}" = "true" ]; then
...
+          " src/util/error.cpp"\
+          " src/util/getuniquepath.cpp"\
+          " src/util/hasher.cpp"\
+          " src/util/message.cpp"\
+          " src/util/moneystr.cpp"\
+          " src/util/serfloat.cpp"\
+          " src/util/spanparsing.cpp"\
+          " src/util/strencodings.cpp"\
+          " src/util/syserror.cpp"\
+          " src/util/url.cpp"\
```
Comment (hebasto) [inline ci/test/06_script_b.sh:60]: Actually, this PR affects more files, e.g., `asmap.cpp`, `readwritefile.cpp` etc. Should they be listed here as well?

Comment (fanquake) [reply ci/test/06_script_b.sh:61]: IWYU generates spurious output for some .cpp files, which would require adding headers that I'm not sure are correct, so I've excluded them for now.

**Review by hebasto — APPROVED** [review]
ACK 07f2c25d04c39a0074e1d9ee1b24b3e359c8153f, I have reviewed the code and it looks OK, I agree it can be merged.

Suggesting to apply `clang-format-diff.py` as well.

**Pull request merged**